Solved the task

Likh-Alex opened the pull request:
Created :
- `ManufacturerDao` interface and `ManufacturerDaoImpl` class
- `Storage` class for storing `Manufacturer` objects
-  `NoSuchManufacturerException` to use with `update` method of `ManufacturerDaoImpl`
- Annotations `Dao` , `Service` , `Inject`
- models `Manufacturer` , `Car` , `Driver`
- `ManufactureService` interface and `ManufacturerServiceImpl` class
- `Application` with `main` method as a driver

Likh-Alex committed:
Solved the task

gigbat commented on pom.xml:
```diff
@@ -9,8 +9,61 @@
+        </maven.checkstyle.plugin.configLocation>
     </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
you can remove it

Likh-Alex replied:
You are absolutely right, no testing though. Done

gigbat commented on src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java:
```diff
@@ -0,0 +1,58 @@
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Optional<Manufacturer> targetId = get(manufacturer.getId());
+        if (targetId.isEmpty()) {
```
as i know, you can use isPresent() into Optional

gigbat replied:
or when you will return, just call the method orElseThrow()

Likh-Alex replied:
You are absolutely right about `isPresent` - but I wanted to `throw` at the very first step and then proceed if `targetId` is not empty.

Likh-Alex replied:
I have rewritten with `isPresent` - I'll give it a try)

gigbat commented on src/main/java/taxi/db/Storage.java:
```diff
@@ -0,0 +1,17 @@
+    private static Long manufacturerId = 0L;
+
+    public static void addManufacturer(Manufacturer manufacturer) {
+        manufacturerId++;
+        manufacturer.setId(manufacturerId);
```
if you want, you can do so:

```
manufacturer.setId(manufacturerId++);
```

Likh-Alex replied:
That looks good to me) thank you)

gigbat commented on src/main/java/taxi/model/Car.java:
```diff
@@ -0,0 +1,49 @@
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(String model, Manufacturer manufacturer) {
```
check for null

Likh-Alex replied:
For future use)

gigbat commented on src/main/java/taxi/model/Car.java:
```diff
@@ -0,0 +1,49 @@
+        return model;
+    }
+
+    public void setModel(String model) {
```
check for null

Likh-Alex replied:
I will change in future, promise)

gigbat commented on src/main/java/taxi/model/Car.java:
```diff
@@ -0,0 +1,49 @@
+        return manufacturer;
+    }
+
+    public void setManufacturer(Manufacturer manufacturer) {
```
same

Likh-Alex replied:
same promise here)

gigbat commented on src/main/java/taxi/model/Driver.java:
```diff
@@ -0,0 +1,36 @@
+    private String name;
+    private String licenseNumber;
+
+    public Driver(String name, String licenseNumber) {
```
check for null everywhere :)

Likh-Alex replied:
I will change everywhere - on next homework)

gigbat reviewed (changes requested):
Good job 👍

Likh-Alex committed:
Refactoring

RomanDubovskyi commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
-target/*
\ No newline at end of file
+target/*
+/src/main/resources/
```
do we need this folder here, why?

Likh-Alex replied:
No) In scope of these project we don't need, accidentally added it to `gitignore`.

RomanDubovskyi commented on .travis.yml:
```diff
@@ -0,0 +1,8 @@
+  - mvn clean verify
+cache:
+  directories:
+    - $HOME/.m2
```
empty line missing

Likh-Alex replied:
Added an empty line in the end

RomanDubovskyi commented on pom.xml:
```diff
@@ -9,8 +9,53 @@
+        </pluginManagement>
+    </build>
+
+
```
same here with the empty line

Likh-Alex replied:
removed redundant empty lines and added empty in the end of the file

RomanDubovskyi commented on src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java:
```diff
@@ -0,0 +1,57 @@
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers
+                .stream()
+                .filter(man -> man.getId().equals(id))
```
```suggestion
                .filter(m -> m.getId().equals(id))
```
this word translation may confuse when reading the code

Likh-Alex replied:
refactored, now the context looks clearer. Thank you

RomanDubovskyi commented on src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java:
```diff
@@ -0,0 +1,57 @@
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Optional<Manufacturer> targetManufacturer = get(manufacturer.getId());
```
we don't need to call get here, because we have fully formed object as method input param

Likh-Alex replied:
refactored to use without get()

RomanDubovskyi commented on src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java:
```diff
@@ -0,0 +1,57 @@
+        Optional<Manufacturer> targetManufacturer = get(manufacturer.getId());
+        for (Manufacturer storageManufacturer : Storage.manufacturers) {
+            if (targetManufacturer.isPresent()) {
+                if (storageManufacturer.getId().equals(targetManufacturer.get().getId())) {
```
use for i loop or stream and method list.set() when you found the needed index instead. This way if any field has been changed or you added a field you won't need to rewrite this code

Likh-Alex replied:
refactored to use an I loop with `list.set()` method. Looks much better now, thank you.

RomanDubovskyi commented on src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java:
```diff
@@ -0,0 +1,57 @@
+
+    @Override
+    public boolean delete(Long id) {
+        Optional<Manufacturer> manufacturer = get(id);
```
use here removeIf method

Likh-Alex replied:
Refactored , that is a very handy method, thank you.

Likh-Alex committed:
- Code style refactoring
- refactored method `update` to use `for i` loop with `List.set()`
- refactored delete to use `List.removeIf()` method

RomanDubovskyi reviewed (approved):
nice, good job 👍
